pkg/pub_integration/lib/src/screenshot_utils.dart

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+import 'package:puppeteer/puppeteer.dart';
+
+// Default screen with 16:10 ratio.
+final desktopDeviceViewport = DeviceViewport(width: 1280, height: 800);
+
+final _screenshotDir = Platform.environment['PUB_SCREENSHOT_DIR'];
+final _isScreenshotDirSet =
+    _screenshotDir != null && _screenshotDir!.isNotEmpty;
+
+// Set this variable to enable screenshot files to be updated with new takes.
+// The default is to throw an exception to prevent accidental overrides from
+// separate tests.
+final _allowScreeshotUpdates =
+    Platform.environment['PUB_SCREENSHOT_UPDATE'] == '1';
+
+// Note: The default values are the last, so we don't need reset
+//       the original values after taking the screenshots.
+final _themes = ['dark', 'light'];
+final _viewports = {
+  'mobile': DeviceViewport(width: 400, height: 800),
+  'tablet': DeviceViewport(width: 768, height: 1024),
```

Comment:
Might I propose we don't do `tablet` (just because of the number of screenshots)? Also don't those normally have much higher resolution (same with phones).

I'm assuming this is more about aspect ratio?

Reply:
Why don't do tablet? I think it is a good way to check how the site behaves in narrower screen, and tbh. I don't think we will manually check tablet screens, but may catch if something degrades.

Reply:
> I'm assuming this is more about aspect ratio?

Mostly the width.

Reply:
I'm mostly thinking that we already have a LOT of screenshots, so I don't mind it, I'm only concerned that it's hard to get an overview.

Reply:
Let's keep them for now, in the hope they are useful. If they are not, create too much noise, we may remove them or ignore them in most comparison reports.